clicache/src/Serializable.cpp

```diff
@@ -273,35 +273,22 @@ namespace Apache
         return (Internal::EnumInfo^)SafeUMSerializableConvertGeneric(sPtr);
       }
 
-      void Serializable::RegisterPdxType(PdxTypeFactoryMethod^ creationMethod)
+      Object^ Serializable::CreateObject(String^ className, Type^ type)
       {
-        if (creationMethod == nullptr) {
-          throw gcnew IllegalArgumentException("Serializable.RegisterPdxType(): "
-                                               "null PdxTypeFactoryMethod delegate passed");
-        }
-        IPdxSerializable^ obj = creationMethod();
-        PdxDelegateMap[obj->GetType()->FullName] = creationMethod;
-        Log::Debug("RegisterPdxType: class registered: " + obj->GetType()->FullName);
-      }
-
-      Object^ Serializable::CreateObject(String^ className)
-      {
-
-        Object^ retVal = CreateObjectEx(className);
+        Object^ retVal = CreateObjectEx(className, type);
 
         if (retVal == nullptr)
         {
-          Type^ t = GetType(className);
-          if (t)
+          if (type)
           {
-            retVal = t->GetConstructor(Type::EmptyTypes)->Invoke(nullptr);
+            retVal = type->GetConstructor(Type::EmptyTypes)->Invoke(nullptr);
             return retVal;
           }
         }
         return retVal;
       }
 
-      Object^ Serializable::CreateObjectEx(String^ className)
+      Object^ Serializable::CreateObjectEx(String^ className, Type^ type)
```

Comment:
Why is this taking `Type^` parameter now?

Reply:
the underlying code previously called `Serializable::GetType` which was moved to `TypeRegistry` and made instance based... now the call sites for `Serializable::CreateObject`  call `cache->TypeRegistry->GetType` inline...